go/libkb/log_send.go

```diff
@@ -239,6 +243,81 @@ func findFirstNewline(b []byte) []byte {
 	return b[(index + 1):]
 }
 
+func appendError(collected []byte, err error) []byte {
+	return append(collected, []byte(fmt.Sprintf("\nERROR READING LOGS:\n%#v\n", err))...)
+}
+
+func readBytesFromPipe(log logger.Logger, pipe io.ReadCloser, numBytes int) ([]byte, error) {
```

Comment:
Any reason to write this function by hand as opposed to using `ioutil.ReadAll`?

If you're worried about the overall size, you could wrap the reader in an `io.LimitedReader` first so it won't go over numBytes.

Reply:
I did look at `LimitReader`. The crucial difference here is that in the case where we have more bytes than we're supposed to, we want to get the _tail_ of the logs, rather than the head. The version that reads straight from files can just seek to a known offset, which is nice, but `journalctl` just gives us a pipe.

The only other option I know of is to use the `journalctl --reverse` flag, which makes the lines come out most-recent-first. I could use that with `LimitReader`, and then split-and-reverse the lines after the fact? I was worried that would be slow and memory intensive, though. (Is it really ok to call `strings.Split` on a 128 MB string? That's literally a million allocations at once 😄 )

Reply:
Ok, makes sense re `LimitedReader`.

Still seems like `ioutil.ReadAll` would work fine...you're concerned that we have some huge log lines that are going to eat up memory?

Reply:
Yeah I was worried about that. Maybe something like this would be good enough:

1) Ask for a number of lines based on 150 bits per line.
2) Put a `LimitReader` on the whole thing for 2x the requested size.
3) If we hit the limit, put a huge all caps warning at the bottom of the log.
4) Assume (3) will never happen :)

I'll try that and see if it simplifies this whole thing.